harvester/core/migrations/0027_query_annotations.py

```diff
@@ -0,0 +1,46 @@
+# Generated by Django 3.2.8 on 2021-11-24 11:16
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('core', '0026_alter_objectiveproperty_property'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Query',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('query', models.CharField(db_index=True, max_length=255)),
+                ('created_at', models.DateTimeField(auto_now_add=True)),
+                ('modified_at', models.DateTimeField(auto_now=True)),
+            ],
+            options={
+                'verbose_name_plural': 'queries',
+            },
+        ),
+        migrations.CreateModel(
+            name='QueryRanking',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('subquery', models.CharField(db_index=True, max_length=255)),
```

Comment:
waarvoor is subquery nodig?

Reply:
Soms zijn er bepaalde resultaten niet. Dus iemand zoekt naar "zeeleven" en verwacht een bepaald artikel, maar die staat niet in de top 10. Vervolgens zoeken ze naar "walvis" en dan vinden ze wel dat ene artikel, dat ze zochten. De "main query" is dan "zeeleven" en de "subquery" wordt "walvis". Vervolgens krijgen onze instellingen strafpunten van het eval algoritme totdat dit ene "walvis" artikel ook verschijnt.

Het is een truc van Jelmer om "presentation bias" effecten te verminderen. De presentation bias betekent zoveel dat als een resultaat nooit gepresenteerd wordt aan een gebruiker de relevantie van dat resultaat niet gekend kan zijn en de gepresenteerde resultaten dus een bias in hun voordeel hebben. De resultaten zijn met andere woorden goed omdat ze gepresenteerd worden, niet omdat ze daadwerkelijk goede resultaten zijn. Althans dat is het risico van presentation bias.

Reply:
Sorry, beetje onduidelijk vraag. Ik snap wat het idee achter subquery is en ik ben het eens dat we die optie moeten bieden, alleen snap ik niet waarom we het moeten opslaan 😉 . Uiteindelijk is het niet nodig voor de search eval toch?

Reply:
Als je mensen wilt terugleiden naar wat ze ooit hebben gedaan, dan zul je het moeten opslaan. Dat is wat de oude tool deed. Het is voor de eval inderdaad niet noodzakelijk. Kunnen we het zo in de standup over hebben.